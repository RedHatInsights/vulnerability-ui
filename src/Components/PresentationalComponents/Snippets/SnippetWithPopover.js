import { Popover, Stack, StackItem, TextContent, Tooltip, Button } from '@patternfly/react-core';
import { ServerAltIcon } from '@patternfly/react-icons';
import propTypes from 'prop-types';
import React from 'react';
import { STATUS_OPTIONS } from '../../../Helpers/constants';
import messages from '../../../Messages';
import { FormattedMessage } from 'react-intl';
import Label from './Label';

const SnippetWithPopover = props => {
    const { type, row } = props;
    const {
        business_risk_text: businessRiskJustification,
        cve_status_id: cveStatusId,
        status_text: statusJustification,
        status_id: systemStatusId
    } = row.attributes;
    const hasDefaultStatus = systemStatusId === 0 && cveStatusId === 0 && statusJustification === null;

    const BusinessRiskPopoverContent = (
        <StackItem>
            <Label>
                <FormattedMessage {...messages.justificationNote} />
            </Label>
            {businessRiskJustification || '--'}
        </StackItem>
    );

    const systemStatusOption = STATUS_OPTIONS.find(option => option.value === systemStatusId.toString());

    const StatusPopoverContent = (
        <StackItem>
            <Label>
                <FormattedMessage {...messages.cveSystemPairStatus} />
            </Label>

            {systemStatusOption?.label || '--'}

            <Label className="pf-u-mt-sm">
                <FormattedMessage {...messages.justificationNote} />
            </Label>
            {statusJustification || '--'}
        </StackItem>
    );

    const statusOption = STATUS_OPTIONS.find(option => option.value === cveStatusId.toString());

    const popoverContent = (
        <Stack>
            {type === 0 ? BusinessRiskPopoverContent : StatusPopoverContent}
            <StackItem>
                <TextContent>
                    <hr className={'splitter pf-u-mb-md pf-u-mt-md'} />
                </TextContent>
                <Label className="pf-u-mt-sm">
                    <FormattedMessage {...messages.cveStatus} />
                </Label>
            </StackItem>
            <StackItem>
                {statusOption?.label || '--'}
            </StackItem>
        </Stack>
    );

    const buttonStyle = {
        wordBreak: 'normal',
        color: hasDefaultStatus ? 'black' : '',
        whiteSpace: 'unset',
        maxWidth: '131px',
        textAlign: 'left',
        padding: '0px'
    };

    return (
        <Popover bodyContent={popoverContent}
            id="popover"
            aria-label={'Business risk popover'}
            position="left"
        >

            <Button variant="link"
                isDisabled={hasDefaultStatus}
                style={buttonStyle}>
                {cveStatusId !== systemStatusId ? (
                    <Tooltip content={<FormattedMessage {...messages.onlyThisSystemCvePair} />}>
                        <ServerAltIcon className="pf-u-m-l"/>
                    </Tooltip>
                ) : ('')}
                {' '}{systemStatusOption?.label || '--'}
            </Button>
        </Popover>
    );
};

SnippetWithPopover.propTypes = {
    type: propTypes.number,
    row: propTypes.object
};
export default SnippetWithPopover;
