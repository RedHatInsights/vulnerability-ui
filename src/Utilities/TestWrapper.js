import propTypes from 'prop-types';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';

const TestWrapper = ({ children, store }) => {
    return (
        <IntlProvider locale="en">
            { store ?
                <Provider store={store}>
                    <Router>
                        {children}
                    </Router>
                </Provider> : children
            }
        </IntlProvider>
    );
};

TestWrapper.propTypes = {
    children: propTypes.node,
    store: propTypes.object
};

export default TestWrapper;
