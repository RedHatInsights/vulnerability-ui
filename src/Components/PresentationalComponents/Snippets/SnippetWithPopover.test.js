import React from 'react';
import SnippetWithPopover from './SnippetWithPopover';
import { render, screen } from '@testing-library/react';
import TestWrapper from '../../../Utilities/TestWrapper';

describe('SnippetWithPopover component', () => {
    it('should render with business risk', () => {
        const rowData = {
            attributes: {
                business_risk_text: "Not Defined",
                business_risk_id: 0,
                cve_status_id: 0,
                description: "desc",
                impact: "Moderate",
                public_date: "2019-09-13T00:00:00+00:00",
                status: "In review",
                status_id: 1,
                status_text: "justification",
                synopsis: "CVE-2019-15031"
            },
            id: "CVE-2019-15031",
            type: "cve"
        };

        const { asFragment } = render(
            <TestWrapper>
                <SnippetWithPopover type={0} row={rowData}/>
            </TestWrapper>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should render with default status', () => {
        const rowData = {
            attributes: {
                business_risk: "Not Defined",
                business_risk_id: 0,
                cve_status_id: 0,
                description: "desc",
                impact: "Moderate",
                public_date: "2019-09-13T00:00:00+00:00",
                status: "Not reviewed",
                status_id: 0,
                status_text: null,
                synopsis: "CVE-2019-15031"
            },
            id: "CVE-2019-15031",
            type: "cve"
        };

        const { asFragment } = render(
            <TestWrapper>
                <SnippetWithPopover type={1} row={rowData}/>
            </TestWrapper>
        );

        expect(screen.getByText('Not reviewed')).toBeTruthy();
    });

    it('should render with non-default status and -- in place of missing or invalid data', () => {
        const rowData = {
            attributes: {
                business_risk: "Not Defined",
                business_risk_id: 0,
                cve_status_id: 20, // invalid option
                description: "desc",
                impact: "Moderate",
                public_date: "2019-09-13T00:00:00+00:00",
                status: null, // replaced by --
                status_id: 20, // invalid option
                status_text: null, // replaced by --
                synopsis: "CVE-2019-15031"
            },
            id: "CVE-2019-15031",
            type: "cve"
        };

        render(
            <TestWrapper>
                <SnippetWithPopover type={1} row={rowData}/>
            </TestWrapper>
        );

        expect(screen.getByText('--')).toBeTruthy();
    });
});
