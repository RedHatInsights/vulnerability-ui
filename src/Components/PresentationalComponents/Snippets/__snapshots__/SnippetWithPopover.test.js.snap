// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnippetWithPopover component should render with business risk 1`] = `
<SnippetWithPopover
  row={
    Object {
      "attributes": Object {
        "business_risk": "Not Defined",
        "business_risk_id": 0,
        "cve_status_id": 0,
        "description": "desc",
        "impact": "Moderate",
        "public_date": "2019-09-13T00:00:00+00:00",
        "status": "In review",
        "status_id": 1,
        "status_text": "justification",
        "synopsis": "CVE-2019-15031",
      },
      "id": "CVE-2019-15031",
      "type": "cve",
    }
  }
  type={0}
>
  <Popover
    aria-label="Business risk popover"
    bodyContent={
      <Stack>
        <StackItem>
          <Label>
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Justification note"
              description="Justification note label for general usage"
              id="justificationNote"
            />
          </Label>
          --
        </StackItem>
        <StackItem>
          <TextContent>
            <hr
              className="splitter pf-u-mb-md pf-u-mt-md"
            />
          </TextContent>
          <Label
            className="pf-u-mt-sm"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="CVE status"
              description="Cve status label for general usage"
              id="cveStatus"
            />
          </Label>
        </StackItem>
        <StackItem>
          Not reviewed
        </StackItem>
      </Stack>
    }
    id="popover"
    position="left"
  >
    <Popper
      appendTo={[Function]}
      distance={25}
      enableFlip={true}
      flipBehavior={
        Array [
          "top",
          "right",
          "bottom",
          "left",
          "top",
          "right",
          "bottom",
        ]
      }
      isVisible={false}
      onDocumentClick={[Function]}
      onDocumentKeyDown={[Function]}
      onTriggerClick={[Function]}
      onTriggerEnter={[Function]}
      placement="left"
      popper={
        <FocusTrap
          active={false}
          aria-describedby="popover-popover-body"
          aria-label="Business risk popover"
          aria-modal="true"
          className="pf-c-popover"
          focusTrapOptions={
            Object {
              "clickOutsideDeactivates": true,
              "fallbackFocus": [Function],
              "returnFocusOnDeactivate": true,
            }
          }
          onMouseDown={[Function]}
          paused={false}
          preventScrollOnDeactivate={true}
          role="dialog"
          style={
            Object {
              "maxWidth": null,
              "minWidth": null,
              "opacity": 0,
              "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
            }
          }
        >
          <PopoverArrow />
          <PopoverContent>
            <PopoverCloseButton
              aria-label="Close"
              onClose={[Function]}
            />
            <PopoverBody
              id="popover-popover-body"
            >
              <Stack>
                <StackItem>
                  <Label>
                    <Memo(MemoizedFormattedMessage)
                      defaultMessage="Justification note"
                      description="Justification note label for general usage"
                      id="justificationNote"
                    />
                  </Label>
                  --
                </StackItem>
                <StackItem>
                  <TextContent>
                    <hr
                      className="splitter pf-u-mb-md pf-u-mt-md"
                    />
                  </TextContent>
                  <Label
                    className="pf-u-mt-sm"
                  >
                    <Memo(MemoizedFormattedMessage)
                      defaultMessage="CVE status"
                      description="Cve status label for general usage"
                      id="cveStatus"
                    />
                  </Label>
                </StackItem>
                <StackItem>
                  Not reviewed
                </StackItem>
              </Stack>
            </PopoverBody>
          </PopoverContent>
        </FocusTrap>
      }
      popperMatchesTriggerWidth={false}
      positionModifiers={
        Object {
          "bottom": "pf-m-bottom",
          "left": "pf-m-left",
          "right": "pf-m-right",
          "top": "pf-m-top",
        }
      }
      trigger={
        <Button
          isDisabled={false}
          style={
            Object {
              "color": "",
              "maxWidth": "131px",
              "padding": "0px",
              "textAlign": "left",
              "whiteSpace": "unset",
              "wordBreak": "normal",
            }
          }
          variant="link"
        >
          <Tooltip
            content={
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Applies only to this specific system and CVE."
                description="Status tooltip content in exposed systems table"
                id="cveHelper.onlyThisSystemCvePair"
              />
            }
          >
            <ServerAltIcon
              className="pf-u-m-l"
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            />
          </Tooltip>
           
          In review
        </Button>
      }
      zIndex={9999}
    >
      <FindRefWrapper
        onFoundRef={[Function]}
      >
        <Button
          isDisabled={false}
          style={
            Object {
              "color": "",
              "maxWidth": "131px",
              "padding": "0px",
              "textAlign": "left",
              "whiteSpace": "unset",
              "wordBreak": "normal",
            }
          }
          variant="link"
        >
          <button
            aria-disabled={false}
            aria-label={null}
            className="pf-c-button pf-m-link"
            data-ouia-component-id="OUIA-Generated-Button-link-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe={true}
            disabled={false}
            role={null}
            style={
              Object {
                "color": "",
                "maxWidth": "131px",
                "padding": "0px",
                "textAlign": "left",
                "whiteSpace": "unset",
                "wordBreak": "normal",
              }
            }
            type="button"
          >
            <Tooltip
              content={
                <Memo(MemoizedFormattedMessage)
                  defaultMessage="Applies only to this specific system and CVE."
                  description="Status tooltip content in exposed systems table"
                  id="cveHelper.onlyThisSystemCvePair"
                />
              }
            >
              <Popper
                appendTo={[Function]}
                distance={15}
                enableFlip={true}
                flipBehavior={
                  Array [
                    "top",
                    "right",
                    "bottom",
                    "left",
                    "top",
                    "right",
                    "bottom",
                  ]
                }
                isVisible={false}
                onBlur={[Function]}
                onDocumentClick={false}
                onDocumentKeyDown={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTriggerEnter={[Function]}
                placement="top"
                popper={
                  <div
                    className="pf-c-tooltip"
                    id="pf-tooltip-2"
                    role="tooltip"
                    style={
                      Object {
                        "maxWidth": null,
                        "opacity": 0,
                        "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
                      }
                    }
                  >
                    <TooltipArrow />
                    <TooltipContent
                      isLeftAligned={false}
                    >
                      <Memo(MemoizedFormattedMessage)
                        defaultMessage="Applies only to this specific system and CVE."
                        description="Status tooltip content in exposed systems table"
                        id="cveHelper.onlyThisSystemCvePair"
                      />
                    </TooltipContent>
                  </div>
                }
                popperMatchesTriggerWidth={false}
                positionModifiers={
                  Object {
                    "bottom": "pf-m-bottom",
                    "left": "pf-m-left",
                    "right": "pf-m-right",
                    "top": "pf-m-top",
                  }
                }
                trigger={
                  <ServerAltIcon
                    className="pf-u-m-l"
                    color="currentColor"
                    noVerticalAlign={false}
                    size="sm"
                  />
                }
                zIndex={9999}
              >
                <FindRefWrapper
                  onFoundRef={[Function]}
                >
                  <ServerAltIcon
                    className="pf-u-m-l"
                    color="currentColor"
                    noVerticalAlign={false}
                    size="sm"
                  >
                    <svg
                      aria-hidden={true}
                      aria-labelledby={null}
                      className="pf-u-m-l"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style={
                        Object {
                          "verticalAlign": "-0.125em",
                        }
                      }
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M1013.1,842.886912 C1005.8,835.586912 997.2,831.986912 987.4,831.986912 L36.6,831.986912 C26.6,831.986912 18.2,835.586912 10.9,842.886912 C3.6,850.186912 0,858.786912 0,868.586912 L0,987.286912 C0,997.286912 3.6,1005.68691 10.7,1013.08691 C18,1020.38691 26.4,1023.98691 36.4,1023.98691 L987.4,1023.98691 C997.4,1023.98691 1005.8,1020.38691 1013.1,1013.08691 C1020.4,1005.78691 1024,997.386912 1024,987.386912 L1024,868.586912 C1024,858.586912 1020.4,850.186912 1013.1,842.886912 Z M960,959.986912 L640,959.986912 L640,895.986912 L960,895.986912 L960,959.986912 Z M1013.1,586.886912 C1005.8,579.586912 997.2,575.986912 987.4,575.986912 L36.6,575.986912 C26.6,575.986912 18.2,579.586912 10.9,586.886912 C3.6,594.186912 0,602.786912 0,612.586912 L0,731.286912 C0,741.286912 3.6,749.686912 10.7,757.086912 C18,764.386912 26.4,767.986912 36.4,767.986912 L987.4,767.986912 C997.4,767.986912 1005.8,764.386912 1013.1,757.086912 C1020.4,749.786912 1024,741.386912 1024,731.386912 L1024,612.586912 C1024,602.586912 1020.4,594.186912 1013.1,586.886912 Z M960,703.986912 L384,703.986912 L384,639.986912 L960,639.986912 L960,703.986912 Z M1013.1,330.886912 C1005.8,323.586912 997.2,319.986912 987.4,319.986912 L36.6,319.986912 C26.6,319.986912 18.2,323.586912 10.9,330.886912 C3.6,338.186912 0,346.786912 0,356.586912 L0,475.286912 C0,485.286912 3.6,493.686912 10.7,501.086912 C18,508.386912 26.4,511.986912 36.4,511.986912 L987.4,511.986912 C997.4,511.986912 1005.8,508.386912 1013.1,501.086912 C1020.4,493.786912 1024,485.386912 1024,475.386912 L1024,356.586912 C1024,346.586912 1020.4,338.186912 1013.1,330.886912 Z M960,447.986912 L768,447.986912 L768,383.986912 L960,383.986912 L960,447.986912 Z M0,245.186912 C0,235.986912 1.2,234.686912 9.6,226.486912 C9.6,226.486912 151.1,86.3869118 161.6,74.1869118 C172.1,61.7869118 185.5,64.1869118 185.5,64.1869118 L840.6,64.1869118 C840.6,64.1869118 852.4,62.9869118 863.1,74.5869118 C873.6,86.1869118 1013.5,226.386912 1013.5,226.386912 C1021.9,234.786912 1024,236.086912 1024,245.186912 C1024,254.786912 1012.3,255.986912 1000.7,255.986912 L22.1,255.986912 C10.8,256.086912 0,254.386912 0,245.186912 L0,245.186912 Z"
                      />
                    </svg>
                  </ServerAltIcon>
                </FindRefWrapper>
              </Popper>
            </Tooltip>
             
            In review
          </button>
        </Button>
      </FindRefWrapper>
    </Popper>
  </Popover>
</SnippetWithPopover>
`;

exports[`SnippetWithPopover component should render with default status 1`] = `
<SnippetWithPopover
  row={
    Object {
      "attributes": Object {
        "business_risk": "Not Defined",
        "business_risk_id": 0,
        "cve_status_id": 0,
        "description": "desc",
        "impact": "Moderate",
        "public_date": "2019-09-13T00:00:00+00:00",
        "status": "Not reviewed",
        "status_id": 0,
        "status_text": null,
        "synopsis": "CVE-2019-15031",
      },
      "id": "CVE-2019-15031",
      "type": "cve",
    }
  }
  type={1}
>
  <Popover
    aria-label="Business risk popover"
    bodyContent={
      <Stack>
        <StackItem>
          <Label>
            <Memo(MemoizedFormattedMessage)
              defaultMessage="CVE-system pair status"
              description="Cve system pair status label for general usage"
              id="cveSystemPairStatus"
            />
          </Label>
          Not reviewed
          <Label
            className="pf-u-mt-sm"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Justification note"
              description="Justification note label for general usage"
              id="justificationNote"
            />
          </Label>
          --
        </StackItem>
        <StackItem>
          <TextContent>
            <hr
              className="splitter pf-u-mb-md pf-u-mt-md"
            />
          </TextContent>
          <Label
            className="pf-u-mt-sm"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="CVE status"
              description="Cve status label for general usage"
              id="cveStatus"
            />
          </Label>
        </StackItem>
        <StackItem>
          Not reviewed
        </StackItem>
      </Stack>
    }
    id="popover"
    position="left"
  >
    <Popper
      appendTo={[Function]}
      distance={25}
      enableFlip={true}
      flipBehavior={
        Array [
          "top",
          "right",
          "bottom",
          "left",
          "top",
          "right",
          "bottom",
        ]
      }
      isVisible={false}
      onDocumentClick={[Function]}
      onDocumentKeyDown={[Function]}
      onTriggerClick={[Function]}
      onTriggerEnter={[Function]}
      placement="left"
      popper={
        <FocusTrap
          active={false}
          aria-describedby="popover-popover-body"
          aria-label="Business risk popover"
          aria-modal="true"
          className="pf-c-popover"
          focusTrapOptions={
            Object {
              "clickOutsideDeactivates": true,
              "fallbackFocus": [Function],
              "returnFocusOnDeactivate": true,
            }
          }
          onMouseDown={[Function]}
          paused={false}
          preventScrollOnDeactivate={true}
          role="dialog"
          style={
            Object {
              "maxWidth": null,
              "minWidth": null,
              "opacity": 0,
              "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
            }
          }
        >
          <PopoverArrow />
          <PopoverContent>
            <PopoverCloseButton
              aria-label="Close"
              onClose={[Function]}
            />
            <PopoverBody
              id="popover-popover-body"
            >
              <Stack>
                <StackItem>
                  <Label>
                    <Memo(MemoizedFormattedMessage)
                      defaultMessage="CVE-system pair status"
                      description="Cve system pair status label for general usage"
                      id="cveSystemPairStatus"
                    />
                  </Label>
                  Not reviewed
                  <Label
                    className="pf-u-mt-sm"
                  >
                    <Memo(MemoizedFormattedMessage)
                      defaultMessage="Justification note"
                      description="Justification note label for general usage"
                      id="justificationNote"
                    />
                  </Label>
                  --
                </StackItem>
                <StackItem>
                  <TextContent>
                    <hr
                      className="splitter pf-u-mb-md pf-u-mt-md"
                    />
                  </TextContent>
                  <Label
                    className="pf-u-mt-sm"
                  >
                    <Memo(MemoizedFormattedMessage)
                      defaultMessage="CVE status"
                      description="Cve status label for general usage"
                      id="cveStatus"
                    />
                  </Label>
                </StackItem>
                <StackItem>
                  Not reviewed
                </StackItem>
              </Stack>
            </PopoverBody>
          </PopoverContent>
        </FocusTrap>
      }
      popperMatchesTriggerWidth={false}
      positionModifiers={
        Object {
          "bottom": "pf-m-bottom",
          "left": "pf-m-left",
          "right": "pf-m-right",
          "top": "pf-m-top",
        }
      }
      trigger={
        <Button
          isDisabled={true}
          style={
            Object {
              "color": "black",
              "maxWidth": "131px",
              "padding": "0px",
              "textAlign": "left",
              "whiteSpace": "unset",
              "wordBreak": "normal",
            }
          }
          variant="link"
        >
          
           
          Not reviewed
        </Button>
      }
      zIndex={9999}
    >
      <FindRefWrapper
        onFoundRef={[Function]}
      >
        <Button
          isDisabled={true}
          style={
            Object {
              "color": "black",
              "maxWidth": "131px",
              "padding": "0px",
              "textAlign": "left",
              "whiteSpace": "unset",
              "wordBreak": "normal",
            }
          }
          variant="link"
        >
          <button
            aria-disabled={true}
            aria-label={null}
            className="pf-c-button pf-m-link pf-m-disabled"
            data-ouia-component-id="OUIA-Generated-Button-link-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe={true}
            disabled={true}
            role={null}
            style={
              Object {
                "color": "black",
                "maxWidth": "131px",
                "padding": "0px",
                "textAlign": "left",
                "whiteSpace": "unset",
                "wordBreak": "normal",
              }
            }
            tabIndex={null}
            type="button"
          >
             
            Not reviewed
          </button>
        </Button>
      </FindRefWrapper>
    </Popper>
  </Popover>
</SnippetWithPopover>
`;

exports[`SnippetWithPopover component should render with non-default status and -- in place of missing or invalid data 1`] = `
<SnippetWithPopover
  row={
    Object {
      "attributes": Object {
        "business_risk": "Not Defined",
        "business_risk_id": 0,
        "cve_status_id": 20,
        "description": "desc",
        "impact": "Moderate",
        "public_date": "2019-09-13T00:00:00+00:00",
        "status": null,
        "status_id": 20,
        "status_text": null,
        "synopsis": "CVE-2019-15031",
      },
      "id": "CVE-2019-15031",
      "type": "cve",
    }
  }
  type={1}
>
  <Popover
    aria-label="Business risk popover"
    bodyContent={
      <Stack>
        <StackItem>
          <Label>
            <Memo(MemoizedFormattedMessage)
              defaultMessage="CVE-system pair status"
              description="Cve system pair status label for general usage"
              id="cveSystemPairStatus"
            />
          </Label>
          --
          <Label
            className="pf-u-mt-sm"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Justification note"
              description="Justification note label for general usage"
              id="justificationNote"
            />
          </Label>
          --
        </StackItem>
        <StackItem>
          <TextContent>
            <hr
              className="splitter pf-u-mb-md pf-u-mt-md"
            />
          </TextContent>
          <Label
            className="pf-u-mt-sm"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="CVE status"
              description="Cve status label for general usage"
              id="cveStatus"
            />
          </Label>
        </StackItem>
        <StackItem>
          --
        </StackItem>
      </Stack>
    }
    id="popover"
    position="left"
  >
    <Popper
      appendTo={[Function]}
      distance={25}
      enableFlip={true}
      flipBehavior={
        Array [
          "top",
          "right",
          "bottom",
          "left",
          "top",
          "right",
          "bottom",
        ]
      }
      isVisible={false}
      onDocumentClick={[Function]}
      onDocumentKeyDown={[Function]}
      onTriggerClick={[Function]}
      onTriggerEnter={[Function]}
      placement="left"
      popper={
        <FocusTrap
          active={false}
          aria-describedby="popover-popover-body"
          aria-label="Business risk popover"
          aria-modal="true"
          className="pf-c-popover"
          focusTrapOptions={
            Object {
              "clickOutsideDeactivates": true,
              "fallbackFocus": [Function],
              "returnFocusOnDeactivate": true,
            }
          }
          onMouseDown={[Function]}
          paused={false}
          preventScrollOnDeactivate={true}
          role="dialog"
          style={
            Object {
              "maxWidth": null,
              "minWidth": null,
              "opacity": 0,
              "transition": "opacity 300ms cubic-bezier(.54, 1.5, .38, 1.11)",
            }
          }
        >
          <PopoverArrow />
          <PopoverContent>
            <PopoverCloseButton
              aria-label="Close"
              onClose={[Function]}
            />
            <PopoverBody
              id="popover-popover-body"
            >
              <Stack>
                <StackItem>
                  <Label>
                    <Memo(MemoizedFormattedMessage)
                      defaultMessage="CVE-system pair status"
                      description="Cve system pair status label for general usage"
                      id="cveSystemPairStatus"
                    />
                  </Label>
                  --
                  <Label
                    className="pf-u-mt-sm"
                  >
                    <Memo(MemoizedFormattedMessage)
                      defaultMessage="Justification note"
                      description="Justification note label for general usage"
                      id="justificationNote"
                    />
                  </Label>
                  --
                </StackItem>
                <StackItem>
                  <TextContent>
                    <hr
                      className="splitter pf-u-mb-md pf-u-mt-md"
                    />
                  </TextContent>
                  <Label
                    className="pf-u-mt-sm"
                  >
                    <Memo(MemoizedFormattedMessage)
                      defaultMessage="CVE status"
                      description="Cve status label for general usage"
                      id="cveStatus"
                    />
                  </Label>
                </StackItem>
                <StackItem>
                  --
                </StackItem>
              </Stack>
            </PopoverBody>
          </PopoverContent>
        </FocusTrap>
      }
      popperMatchesTriggerWidth={false}
      positionModifiers={
        Object {
          "bottom": "pf-m-bottom",
          "left": "pf-m-left",
          "right": "pf-m-right",
          "top": "pf-m-top",
        }
      }
      trigger={
        <Button
          isDisabled={false}
          style={
            Object {
              "color": "",
              "maxWidth": "131px",
              "padding": "0px",
              "textAlign": "left",
              "whiteSpace": "unset",
              "wordBreak": "normal",
            }
          }
          variant="link"
        >
          
           
          --
        </Button>
      }
      zIndex={9999}
    >
      <FindRefWrapper
        onFoundRef={[Function]}
      >
        <Button
          isDisabled={false}
          style={
            Object {
              "color": "",
              "maxWidth": "131px",
              "padding": "0px",
              "textAlign": "left",
              "whiteSpace": "unset",
              "wordBreak": "normal",
            }
          }
          variant="link"
        >
          <button
            aria-disabled={false}
            aria-label={null}
            className="pf-c-button pf-m-link"
            data-ouia-component-id="OUIA-Generated-Button-link-3"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe={true}
            disabled={false}
            role={null}
            style={
              Object {
                "color": "",
                "maxWidth": "131px",
                "padding": "0px",
                "textAlign": "left",
                "whiteSpace": "unset",
                "wordBreak": "normal",
              }
            }
            type="button"
          >
             
            --
          </button>
        </Button>
      </FindRefWrapper>
    </Popper>
  </Popover>
</SnippetWithPopover>
`;
